### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by D3nnis249**
Thank you for you review,

client/src/components/NavBar.jsx , in this pull request we created the navbar component and put it on the mainpage. then we installed styled components and prepared the navbar for the next tasks.

client/src/globalStyles.js, here we have added global styled to set some default settings like witdth, height, padding and margin

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const Wrapper = styled.div`
+  display: flex;
+  width: 100%;
+  justify-content: center;
+  
+`;
+
+const Main = styled.main`
+  display: flex;
+  width: 70vw;y<
```
Comment (AlexF090) [inline client/src/App.jsx]: We have already noticed the bug and have fixed it.

Comment (mpagels) [reply client/src/App.jsx]: Then why didn't you push it to the branch? The error is still there. ;-)

**Comment by mpagels** [comment]
I will check this pull request only after #10 and #11 are merged into main.

File: client/src/App.css [hunk 2]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (PReichetanz) [inline client/src/App.css]: If you installed styled-components in this PR, why do you keep the App.css?

File: client/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import styled from 'styled-components';
+import Header from './components/Header';
+import NavBar from './components/NavBar';
+import './App.css';
+import data from './data/db';
+import FavoriteIconDeactivated from './icons/favorite_icon_deactivated.svg';
+
+function App() {
+  console.log(data);
```
Comment (PReichetanz) [inline client/src/App.jsx]: Please remove all console.logs before merging.

File: client/src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  return (
+    <Wrapper>
+      <Header />
+      <Main>
+      <StyledUList>
```
Comment (PReichetanz) [inline client/src/App.jsx]: There should be some indentation, is your prettier broken here?

Comment (PReichetanz) [inline client/src/App.jsx]: I am not really a friend of using "styled" when naming styled components, what about "GasStationList" and "GasStationItem"?

File: client/src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  border-radius: 10px;
+  margin-bottom: 4px;
+  padding: 15px;
+  list-style: none;
+  /* border: 1px solid black; */
+`;
+
+//
+//
+
```
Comment (PReichetanz) [inline client/src/App.jsx]:
```suggestion
```

File: client/src/App.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+`;
+
+const Main = styled.main`
+  display: flex;
+  width: 70vw;y<
+  justify-content: center;
+  
+`;
+
+const StyledUList = styled.ul`
```
Comment (PReichetanz) [inline client/src/App.jsx]: See naming issue above.

File: client/src/App.jsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+display: flex;
+margin-top: 120px;
+margin-bottom: 100px;
+flex-direction: column;
+gap: 10px;
+`;
+
+const StyledGasStationItem = styled.li`
```
Comment (PReichetanz) [inline client/src/App.jsx]: Same here.

File: client/src/App.jsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.25), inset 0px 0px 2px rgba(0, 0, 0, 0.25);
+  border-radius: 10px;
+  margin-bottom: 4px;
+  padding: 15px;
+  list-style: none;
```
Comment (PReichetanz) [inline client/src/App.jsx]: Please make sure your list is still accessible when deactivating list-style, e.g. by using the role attribute.

File: client/src/components/GasStation.jsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  const [gasStation, setGasStation] = useState({});
+  const handleDetails = () => {
+    setGasStationDetailsOn(changeState => !changeState);
+  };
+
+//   useEffect(() => {
+//     fetch('../data/db.js' + uuid).then(response =>
+//       response.json().then(data => setGasStation(data))
+//     );
+//   }, []);
```
Comment (PReichetanz) [inline client/src/components/GasStation.jsx:17]: Please remove unused code before creating a PR.

File: client/src/components/Header.jsx [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+const StyledHeader = styled.h1`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 6.25rem;;
+  position: fixed;
+  top: 0;
+  border: 1px solid black;
+
```
Comment (PReichetanz) [inline client/src/components/Header.jsx:19]:
```suggestion
```

File: client/src/components/NavBar.jsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import styled from 'styled-components';
+import StartIcon from '../icons/start_icon.svg'
+import MapIcon from '../icons/map_icon.svg'
+import SettingsIcon from '../icons/settings_icon.svg'
+import FavoriteIcon from '../icons/favorite_icon.svg'
+
+
+
```
Comment (PReichetanz) [inline client/src/components/NavBar.jsx]:
```suggestion
```

File: client/src/components/NavBar.jsx [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+function NavBar() {
+  return (
+    <StyledNavBar>
+      <StyledUnorderedList>
+        <StyledListItem><img src={StartIcon} alt="Start Icon" /><figcaption>Start</figcaption></StyledListItem>
+        <StyledListItem><img src={MapIcon} alt="Karten Icon" /><figcaption>Karte</figcaption></StyledListItem>
+        <StyledListItem><img src={SettingsIcon} alt="Einstellungs Icon" /><figcaption>Einstellungen</figcaption></StyledListItem>
+        <StyledListItem><img src={FavoriteIcon} alt="Favoriten Icon" /><figcaption>Favoriten</figcaption></StyledListItem>
+      </StyledUnorderedList>
+    </StyledNavBar>
```
Comment (PReichetanz) [inline client/src/components/NavBar.jsx]: I would reconsider the naming without styled as mentioned above.

File: client/src/components/NavBar.jsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+const StyledNavBar = styled.nav`
+  display: flex;
+  width: 100%;
+  height: 4.6875rem;
+  position: fixed;
+  bottom: 0;
+  border: 1px solid black;
+
```
Comment (PReichetanz) [inline client/src/components/NavBar.jsx]:
```suggestion
```

File: client/src/components/NavBar.jsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+text-align: center;
+display: flex;
+flex-direction: row;
+align-items: center;
+justify-content: space-around;
+color: #2196F3;
+`;
+
+const StyledListItem = styled.li`
+list-style: none;
```
Comment (PReichetanz) [inline client/src/components/NavBar.jsx]: Please make sure your list is still accessible, see above.

File: client/src/data/db.js [hunk 15]
```diff
@@ -0,0 +1,383 @@
+const data = [
+
```
Comment (PReichetanz) [inline client/src/data/db.js]:
```suggestion
```

File: client/src/data/db.js [hunk 16]
```diff
@@ -0,0 +1,383 @@
...
+    city: 'Röttenbach',
+    latitude: 49.660976409912,
+    longitude: 10.918617248535,
+    openingtimes_json: {},
+    diesel: 1.999,
+    e5: 1.999,
+    e10: 1.949,
+  }
+  ,
+  {
```
Comment (PReichetanz) [inline client/src/data/db.js]:
```suggestion
  },
  {
```

File: client/src/globalStyles.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+  html, body {
+    height: 100%;
+    width: 100%;
+}
+
+body {
+    display: flex;
+    flex-direction: column;
+}
+`;
```
Comment (PReichetanz) [inline client/src/globalStyles.js]: The indentation seems strange here: Why is body not indented?

**Review by PReichetanz — CHANGES_REQUESTED** [review]
Nice work here! I added some comments which came to my mind, please consider them and let me know if you have any questions.

File: client/src/components/NavBar.jsx [hunk 18]
```diff
@@ -1,49 +1,62 @@
...
-display: flex;
-flex-direction: row;
-align-items: center;
-justify-content: space-around;
-color: #2196F3;
+const UnorderedList = styled.ul`
+  width: 100%;
+  text-align: center;
+  display: flex;
+  flex-direction: row;
```
Comment (mbosselmann) [inline client/src/components/NavBar.jsx:51]: `flex-direction: row` is the default of `flex-direction`. Therefore, you don't need to write it.

File: client/src/index.js [hunk 19]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
-import GlobalStyle from './globalStyles';
+import GlobalStyle from './GlobalStyles';
```
Comment (mbosselmann) [inline client/src/index.js:5]: I would name it `GlobalStyles` when the name of your file is also `GlobalStyles`. ;)

**Review by mbosselmann — APPROVED** [review]
Great job! 🚀 Keep it up! 💪 Just two nit-picky things. ;)

**Pull request merged**